### ISSUE TITLE
Exception in thread "OkHttp Dispatcher" com.binance.api.client.exception.Binance

### ISSUE THREAD

**Issue by xezon** [binance-exchange/binance-java-api#24]
I run into this exception very often when using the web socket listeners. What is the correct way to deal with failing sockets? If I understand correctly there is no way to ignore the exception, but just accept the closure of the socket and ultimatively the thread.

But is there any smart way to restart the listener? I don't know which listeners closed. Do I need to resort to thread naming trickery and a thread watch dog?

```
Exception in thread "OkHttp Dispatcher" com.binance.api.client.exception.Binance
ApiException: java.io.EOFException
        at com.binance.api.client.impl.BinanceApiWebSocketListener.onFailure(Bin
anceApiWebSocketListener.java:39)
        at okhttp3.internal.ws.RealWebSocket.failWebSocket(RealWebSocket.java:54
3)
        at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:203
)
        at okhttp3.RealCall$AsyncCall.execute(RealCall.java:141)
        at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.io.EOFException
        at okio.RealBufferedSource.require(RealBufferedSource.java:60)
        at okio.RealBufferedSource.readByte(RealBufferedSource.java:73)
        at okhttp3.internal.ws.WebSocketReader.readHeader(WebSocketReader.java:1
13)
        at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.
java:97)
        at okhttp3.internal.ws.RealWebSocket.loopReader(RealWebSocket.java:262)
        at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:201
)
        ... 5 more
```

**Comment by xezon** [comment]
Note: this pull request provides an opportunity to deal with this problem:

https://github.com/binance-exchange/binance-java-api/pull/34

**Comment by datalorax** [comment]
#49 Also looks to address this issue.

**Comment by joaopsilva** [comment]
Fixed in #49.
